vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt

```diff
@@ -252,98 +252,209 @@ class OidcSiopWallet(
      * Creates the authentication response from the RP's [params]
      */
     suspend fun createAuthnResponseParams(
-        params: AuthenticationRequestParameters
+        params: AuthenticationRequestParametersFrom<*>
     ): KmmResult<AuthenticationResponseParameters> {
-        if (params.clientIdScheme == OpenIdConstants.ClientIdSchemes.REDIRECT_URI
-            && (params.clientMetadata == null && params.clientMetadataUri == null)
+        //TODO? more sanity checks now that we have mor info
+        if (params.parameters.clientIdScheme == OpenIdConstants.ClientIdSchemes.REDIRECT_URI
+            && (params.parameters.clientMetadata == null && params.parameters.clientMetadataUri == null)
         ) {
             return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
                 .also { Napier.w("client_id_scheme is redirect_uri, but metadata is not set") }
         }
+
+        //TODO: was the JWsSigned already cryptographically verified? esp. th x5c?
+        if ((params.parameters.clientIdScheme == OpenIdConstants.ClientIdSchemes.X509_SAN_DNS)
+            || (params.parameters.clientIdScheme == OpenIdConstants.ClientIdSchemes.X509_SAN_DNS)
+        ) {
+            if (params.parameters.clientMetadata == null
+                || params !is AuthenticationRequestParametersFrom.JwsSigned
+                || params.source.header.certificateChain == null
+                || params.source.header.certificateChain!!.isEmpty()
+            ) return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
+                .also {
+                    Napier.w(
+                        "client_id_scheme is ${
+                            params.parameters.clientIdScheme
+                        }, but metadata is not set and no x5c certificate chain is present in the original authn request"
+                    )
+                }
+            else { //basic checks done
+                val leaf = params.source.header.certificateChain!!.leaf
+                if (leaf.tbsCertificate.extensions == null || leaf.tbsCertificate.extensions!!.isEmpty()) {
+                    return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
+                        .also {
```

Comment:
please reformat the logging statements to be ... somehow on one lin, so we can skim through the code more easily